lib/dotcom_web/controllers/trip_plan_controller.ex

```diff
@@ -5,305 +5,43 @@ defmodule DotcomWeb.TripPlanController do
 
   use DotcomWeb, :controller
 
-  require Logger
+  import DotcomWeb.Router.Helpers, only: [live_path: 2]
 
-  alias Dotcom.TripPlan.{
-    Itinerary,
-    ItineraryRowList,
-    Leg,
-    NamedPosition,
-    PersonalDetail,
-    Query,
-    RelatedLink,
-    TransitDetail
-  }
-
-  alias Dotcom.TripPlan.Map, as: TripPlanMap
-  alias Routes.Route
+  alias Dotcom.TripPlan.AntiCorruptionLayer
 
   @location_service Application.compile_env!(:dotcom, :location_service)
 
-  @type route_map :: %{optional(Route.id_t()) => Route.t()}
-  @type route_mapper :: (Route.id_t() -> Route.t() | nil)
-
-  plug(:assign_initial_map)
-  plug(:breadcrumbs)
-  plug(:modes)
-  plug(:wheelchair)
-  plug(:meta_description)
-  plug(:assign_params)
-
-  def index(conn, %{"plan" => %{"to" => _to, "from" => _fr} = plan}) do
-    conn
-    |> assign(:expanded, conn.query_params["expanded"])
-    |> render_plan(plan)
-  end
-
-  def index(conn, _params) do
-    render(conn, :index)
-  end
-
-  def from(conn, %{"plan" => _plan} = params) do
-    redirect(conn, to: trip_plan_path(conn, :index, Map.delete(params, "address")))
-  end
-
-  def from(conn, %{
-        "address" => address
-      }) do
-    if String.match?(address, ~r/^(\-?\d+(\.\d+)?),(\-?\d+(\.\d+)?),.*$/) do
-      [latitude, longitude, name] = String.split(address, ",", parts: 3)
-      # Avoid extra geocode call, just use these coordinates
-      destination = %NamedPosition{
-        latitude: String.to_float(latitude),
-        longitude: String.to_float(longitude),
-        name: name,
-        stop: nil
-      }
+  @doc """
+  When visiting /trip-planner/:from/some%20address or /trip-planner/:to/some%20address,
+  we lookup the location and redirect to the trip planner with that location encoded in the query string.
+  """
+  def location(conn, %{"direction" => direction, "query" => query})
+      when direction in ["from", "to"] do
 
-      do_from(conn, destination)
-    else
-      updated_address = check_address(address)
+    case @location_service.geocode(query) do
+      {:ok, [%LocationService.Address{} = location | _]} ->
+        encoded = build_params(direction, location) |> AntiCorruptionLayer.encode()
+        path = live_path(conn, DotcomWeb.Live.TripPlanner)
```

Comment:
Huh, that's interesting that `live_path` works in this context

**question**: if we merge this before releasing the new trip planner, does this resolve to nil and let this proceed to the current `/trip-planner?plan={stuff}`? or will /from/ and /to/ be broken until the new trip planner lands?